### ISSUE TITLE
Target built because of InitialTargets should report it via BuildReason

### ISSUE THREAD

**Issue by KirillOsenkov** [dotnet/msbuild#6775]
Currently when a target is built because it was specified in InitialTargets the BuildReason is passed as None on the TargetStartedEventArgs.

We should add a new enum value on BuildReason and pass that.

```xml
<Project InitialTargets="A" DefaultTargets="Build">

  <Target Name="A" />
  <Target Name="Build" DependsOnTargets="A" />

</Project>
```

![image](https://user-images.githubusercontent.com/679326/130287251-b6dbd10c-fe10-4898-8cc1-ace197aec480.png)

**Comment by KirillOsenkov** [comment]
Unfortunately adding a new enum value might be a breaking change since consumers might throw if they get an enum value they don't support... Need to see if that's a real problem.

**Comment by rainersigwald** [comment]
Adding an enum value isn't on [the list](https://docs.microsoft.com/dotnet/core/compatibility/), so I think it's ok? It would break anyone who did the (completely reasonable! I do it all the time!) "handle `default` in a switch by throwing `NotImplementedException`" strategy though.

**Comment by KirillOsenkov** [comment]
Yup, that's what I was alluding to. I'm guessing we should just try it and see if we break any third party loggers. It's not like we've never broken them before ;P
